src/xenia/base/memory_posix.cc

```diff
@@ -82,16 +88,19 @@ FileMappingHandle CreateFileMappingHandle(const std::filesystem::path& path,
   }
 
   oflag |= O_CREAT;
-  int ret = shm_open(path.c_str(), oflag, 0777);
+  auto full_path = "/" / path;
+  int ret = shm_open(full_path.c_str(), oflag, 0777);
   if (ret > 0) {
     ftruncate64(ret, length);
   }
 
   return ret <= 0 ? nullptr : reinterpret_cast<FileMappingHandle>(ret);
 }
 
-void CloseFileMappingHandle(FileMappingHandle handle) {
-  close((intptr_t)handle);
+void CloseFileMappingHandle(FileMappingHandle handle,
+                            const std::filesystem::path& path) {
+  auto full_path = "/" / path;
```

Comment:
According to http://www.qnx.com/developers/docs/6.5.0/index.jsp?topic=%2Fcom.qnx.doc.neutrino_lib_ref%2Fs%2Fshm_unlink.html, it appears that both close and shm_unlink are needed to properly close and destroy the shared memory object — previously this function would leave the shared memory object created in the system, now it possibly won't be unlinked until the process ends, since an open file descriptor is left.
> This function doesn't affect any references to the shared memory object (i.e. file descriptors or memory mappings). If more than one reference to the shared memory object exists, then the link count is decremented, but the shared memory segment isn't actually removed until you remove all open and map references to it.